### PR TITLE
Ability to delete experiments in Aim UI

### PR DESCRIPTION
**Pull request by fabiovincenzi**
This PR only implements the backend side. The UI side is implemented in https://github.com/G-Research/fasttrackml-ui-aim/pull/6.

### CODE REVIEW

File: pkg/api/mlflow/dao/repositories/experiment.go [hunk 1]
```diff
@@ -107,3 +108,31 @@ func (r ExperimentRepository) Update(ctx context.Context, experiment *models.Exp
 
 	return nil
 }
+
+// Delete removes the existing models.Experiment from the db.
+func (r ExperimentRepository) Delete(ctx context.Context, experiment *models.Experiment) error {
```
Comment (suprjinx) [inline pkg/api/mlflow/dao/repositories/experiment.go:113]: I think these two signatures need to be added to ExperimentRepositoryProvider interface

Comment (fabiovincenzi) [reply pkg/api/mlflow/dao/repositories/experiment.go:117]: Fixed

**Review by suprjinx — DISMISSED** [review]

File: pkg/ui/aim/embed/custom.patch [hunk 2]
```diff
@@ -414,7 +434,123 @@ index 9c9fb38..fb0de77 100644
...
+     analytics.pageView(ANALYTICS_EVENT_KEYS.experiment.tabs.settings.tabView);
+   }, []);
+ 
+@@ -34,7 +56,28 @@ function ExperimentSettingsTab({
+             defaultDescription={description ?? ''}
+             onSave={onSave}
+           />
++          <ActionCard
++            title='Delete Experiment'
++            description='Once you delete an expetiment, there is no going back. Please be certain.'
```
Comment (jgiannuzzi) [inline pkg/ui/aim/embed/custom.patch:531]: typo: expetiment -> experiment

Comment (fabiovincenzi) [reply pkg/ui/aim/embed/custom.patch]: fixed

**Review by suprjinx — APPROVED** [review]

File: pkg/database/database.go [hunk 3]
```diff
@@ -394,13 +394,32 @@ func checkAndMigrateDB(db *DbInstance, migrate bool) error {
 				}); err != nil {
 					return fmt.Errorf("error migrating database to FastTrackML schema 1ce8669664d2: %w", err)
 				}
+				fallthrough
+
+			case "1ce8669664d2":
```
Comment (dsuhinin) [inline pkg/database/database.go:399]: just curious -> how we select migration id?  `5d042539be4f` probably randomly but ....?

**Review by dsuhinin — DISMISSED** [review]
It looks good.
cc: @jgiannuzzi

File: pkg/database/database.go (hunk 3 above)
Comment (fabiovincenzi) [reply pkg/database/database.go:399]: Yep, randomly

File: pkg/ui/aim/embed/custom.patch [hunk 4]
```diff
@@ -638,3 +774,105 @@ index 0fc79c5..867376b 100644
...
++
++const runDetailAppModel = {
++  ...model,
++  initialize,
++  deleteExperiment,
++  onNotificationAdd,
++  onNotificationDelete,
++};
++
++export default runDetailAppModel;
```
Comment (jgiannuzzi) [inline pkg/ui/aim/embed/custom.patch:878]: should this not belong to `aim/web/ui/src/pages/Experiment/ExperimentStore.ts` instead?

File: pkg/database/database.go [hunk 5]
```diff
@@ -394,13 +394,32 @@ func checkAndMigrateDB(db *DbInstance, migrate bool) error {
...
+				}); err != nil {
+					return fmt.Errorf("error migrating database to FastTrackML schema 5d042539be4f: %w", err)
+				}
 
 			default:
 				return fmt.Errorf("unsupported database FastTrackML schema version %s", schemaVersion.Version)
 			}
-
 			log.Info("Database migration done")
-
```
Comment (jgiannuzzi) [inline pkg/database/database.go:403]: nit: leave the blank lines for readability

**Review by jgiannuzzi — COMMENTED** [review]
This is really good, thanks @fabiovincenzi!

I have a couple of nitpicks below about the UI side of things. The backend LGTM 👍

**Comment by dsuhinin** [comment]
> This is really good, thanks @fabiovincenzi!
> 
> I have a couple of nitpicks below about the UI side of things. The backend LGTM 👍

annnnnd we agreed on `integration` tests, right? for any new logic?

File: pkg/ui/aim/embed/custom.patch (hunk 4 above)
Comment (fabiovincenzi) [reply pkg/ui/aim/embed/custom.patch:878]: I was following the approach they use with run deletion in `aim/web/ui/src/pages/RunDetail/RunDetailSettingsTab/RunDetailSettingsTab.tsx`

**Review by dsuhinin — DISMISSED** [review]

**Review by jgiannuzzi — CHANGES_REQUESTED** [review]
Deleting experiments should renumber the associated runs

**Review by suprjinx — APPROVED** [review]

**Review by jgiannuzzi — APPROVED** [review]
LGTM :+1:

**Pull request merged**